File: Siav/SiavWeb/src/main/webapp/scripts/services/rangos-services.js
/*global define*/
'use strict';

define(['siav-module', 'constantes'], function (app) {
    app.service('rangosServices', ['$http', '$q', 'CONSTANTES', function ($http, $q, CONSTANTES) {
    	 var contrato = {
    			 consultar : consultar,
    			 crear : crear,
    			 eliminar : eliminar
    	 };
    	 
    	 function crear(rango){
    		 var request = $http.post(CONSTANTES.SRV.RANGO_CREAR, rango);
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
    	 
    	 function eliminar(rango){
    		 var request = $http.post(CONSTANTES.SRV.RANGO_ELIMINAR, rango);
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
    	 
    	 function consultar(){
    		 var request = $http.get(CONSTANTES.SRV.RANGO_CONSULTAR, {isArray : true});
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
   	 
    	 return contrato;
    }]);
});

File: Siav/SiavWeb/src/main/java/co/com/siav/rest/services/RangosServices.java
package co.com.siav.rest.services;

import java.util.List;

import javax.enterprise.context.RequestScoped;
import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import co.com.siav.bean.RangosBean;
import co.com.siav.entities.Rango;
import co.com.siav.response.MensajeResponse;


@RequestScoped
@Path("rangos/")
public class RangosServices {
	
	@Inject
	private RangosBean bean;
	
	@POST
	@Path("crear")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	public MensajeResponse createRango(Rango request){
		return bean.crear(request);
	}
	
	@POST
	@Path("eliminar")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	public MensajeResponse deleteRango(Rango request){
		return bean.eliminar(request);
	}
	
	@GET
	@Path("consultar")
	@Produces(MediaType.APPLICATION_JSON)
	public List<Rango> getRangos(){
		return bean.consultar();
	}

}
